### PR TITLE
Trello 22 2

### PR DESCRIPTION
**Pull request by ghost**
This PR includes commits that integrate the lyncode [XOAI](https://github.com/lyncode/XOAI-DSpace-Addon-1.8.2) module for DSpace 1.x. It also includes some modifications to the XOAI module to support usage of DOI rather handles as item identifiers.

The XOAI module was used to implement the updates in [trello-22](https://trello.com/c/opgl75VS/22-clean-up-oai-pmh-api) because it relies largely on a metadata pipeline that enables flexibility in configuration using the module's configuration and stylesheet files, and allowed the application updates to be confined to a single module.

This PR has been used to for a rebuild of the dev server at: http://dev.datadryad.org/xoai/request?verb=Identify

Deploying this module has dependencies outside of this PR, including Maven settings and SOLR indexing, which have been documented elsewhere.

There are four commits here, which involve:

ce56c46
Initial import of XOAI module and inclusion of XOAI in dspace build:
- adds dsapce-xoai/ module at root of dspace dir
- adds xoai config values to dspace/config/dspace.cfg 
- add xoai commands to dspace/config/launcher.xml
- adds config and data files to dspace/config/modules/xoai 
- adds dspace/modules/xoai
- updates existing build files: pom.xml, dspace/pom.xml, dspace/modules/pom.xml

Most of this is new files imported from the module.

1febae4
Externalizes some strings to config file. XSL and data changes here are overridden by the next commit.

5895a14
Adds Dryad custom stylesheets and config to XOAI data/config files, to remove handle identifiers.

4f79318
Additional externalization of query parameters to support DOI usage. Cherry-pick of shared testing file. Regression tests.

[Edit: update commit numbers after rebase; testing resources.]

### CODE REVIEW

**Comment by daisieh** [comment]
@rnathanday: how should I test this?

**Comment by ghost** [comment]
@daisieh sorry, I seem to have left out the tests that accompany the module. I will need to update and push before they're available

**Comment by ghost** [comment]
@daisieh The module can be integration tested on dev to confirm that the interface and solr integration are working as expected. An example URL is: http://dev.datadryad.org/xoai/request?verb=Identify

The module exposes endpoints documented here, mutatis mutandis: http://wiki.datadryad.org/Data_Access#OAI-PMH , with two caveats:

 1) The path for this module begins `http://dev.datadryad.org/xoai/...`, ie, with an `xoai`.
 2) DOIs instead of handles are used as identifiers, e.g.: http://dev.datadryad.org/xoai/request?verb=GetRecord&identifier=oai:datadryad.org:doi:10.5061/dryad.12&metadataPrefix=oai_dc

Navigating to an xoai URL with a browser will return a webpage, while navigating programmatically (eg, with curl) will return the expected OAI-PMH XML record.

I expect to have the unit tests pushed to this branch shortly.

File: dspace/config/dspace.cfg [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#
+${}#
```
Comment (daisieh) [inline dspace/config/dspace.cfg]: Is this supposed to be here?

Comment (ghost) [inline dspace/config/dspace.cfg]: No it isn't -- not sure why it doesn't cause an error during build. Thanks for catching that

**Comment by ghost** [comment]
@daisieh There have been regression tests added to the repo that will confirm that the Dryad modifications (dois rather than handles, https://trello.com/c/opgl75VS) are in place. The updated code has been deployed on dev.

**Comment by daisieh** [comment]
As far as I can tell, this code looks fine (I didn't review the code that was ported from Lyncode), and it works as expected. Merging.

**Pull request merged**